adafruit_io/adafruit_io.py

```diff
@@ -40,9 +40,11 @@ def validate_feed_key(feed_key):
     """
     if len(feed_key) > 128:  # validate feed key length
         raise ValueError("Feed key must be less than 128 characters.")
-    if not bool(re.match("^[a-z0-9-]+$", feed_key)):  # validate key naming scheme
+    if not bool(
+        re.match("^[a-z0-9-]+(\\.[a-z0-9-]+)?$", feed_key)
```

Comment:
The double `\` is not needed, you're already escaping the dot with a single `\` .

A double `\\.` here means match the empty string followed by a literal dot, which does not cause any harm here but formally is not correct :)

Reply:
> The double `\` is not needed, you're already escaping the dot with a single `\` .
> 
> A double `\\.` here means match the empty string followed by a literal dot, which does not cause any harm here but formally is not correct :)

hmm.... I think you are not right about that. Could you give it a try? This is what I did:

```python
>>> import re
>>> feed_key = "bad."
>>> bool(re.match("^[a-z0-9-]+(\\.[a-z0-9-]+)?$", feed_key))
False
>>> feed_key = "not.bad"
>>> bool(re.match("^[a-z0-9-]+(\\.[a-z0-9-]+)?$", feed_key))
True
>>> feed_key = "is.b!ad"
>>> bool(re.match("^[a-z0-9-]+(\\.[a-z0-9-]+)?$", feed_key))
False
>>> 
```

Also, this is what I see if I do not use the extra '\':
```
$ pre-commit run --all-files
...
pylint (library code)....................................................Failed
- hook id: pylint
- exit code: 4

************* Module adafruit_io.adafruit_io
adafruit_io/adafruit_io.py:44:30: W1401: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```
Ref: https://stackoverflow.com/questions/19030952/pep8-warning-on-regex-string-in-python-eclipse

Reply:
Well, as I said in this context works anyway with or without the double `\` : 
```
>>> feed_key = "bad."
>>> bool(re.match("^[a-z0-9-]+(\.[a-z0-9-]+)?$", feed_key))
False
>>> feed_key = "not.bad"
>>> bool(re.match("^[a-z0-9-]+(\.[a-z0-9-]+)?$", feed_key))
True
>>> feed_key = "is.b!ad"
>>> feed_key = "is.b!ad"
>>> bool(re.match("^[a-z0-9-]+(\.[a-z0-9-]+)?$", feed_key))
False
```

To satisfy the linter, I think you can just prefix your string with an `r` : re.match(r"^[a-z0-9-]+(\.[a-z0-9-]+)?$") , or yes use the double escape (personally I think that an `r` makes the expression more readable)

Reply:
Done.